executor/simple.go

```diff
@@ -47,11 +47,6 @@ type SimpleExec struct {
 	is        infoschema.InfoSchema
 }
 
-// Next implements Execution Next interface.
-func (e *SimpleExec) Next(ctx context.Context) (Row, error) {
-	return nil, errors.Trace(e.run(ctx))
-}
-
 // NextChunk implements the Executor NextChunk interface.
 func (e *SimpleExec) NextChunk(ctx context.Context, chk *chunk.Chunk) error {
 	return errors.Trace(e.run(ctx))
```

Comment:
We can remove the function `run`,
and put the implementation in NextChunk.

Reply:
```go
    if e.done {
		return nil
	}
	err := e.run(ctx)
	e.done = true
	return errors.Trace(err)
```
Is it?

Reply:
``` go
// NextChunk implements the Executor NextChunk interface.
func (e *SimpleExec) NextChunk(ctx context.Context, chk *chunk.Chunk) (err error) {
	if e.done {
		return nil
	}
	switch x := e.Statement.(type) {
	case *ast.UseStmt:
		err = e.executeUse(x)
	case *ast.FlushStmt:
		err = e.executeFlush(x)
	case *ast.BeginStmt:
		err = e.executeBegin(x)
	case *ast.CommitStmt:
		e.executeCommit(x)
	case *ast.RollbackStmt:
		err = e.executeRollback(x)
	case *ast.CreateUserStmt:
		err = e.executeCreateUser(x)
	case *ast.AlterUserStmt:
		err = e.executeAlterUser(x)
	case *ast.DropUserStmt:
		err = e.executeDropUser(x)
	case *ast.SetPwdStmt:
		err = e.executeSetPwd(x)
	case *ast.KillStmt:
		err = e.executeKillStmt(x)
	case *ast.BinlogStmt:
		// We just ignore it.
		return nil
	case *ast.DropStatsStmt:
		err = e.executeDropStats(x)
	}
	e.done = true
	return errors.Trace(err)
}
```